### PR TITLE
pre-commit: PR112742

### PR DESCRIPTION
**Pull request by dtcxzyw**
Link: https://github.com/llvm/llvm-project/pull/112742
Requested by: @dtcxzyw

### CODE REVIEW

File: bench/arrow/optimized/hashing.cc.ll [hunk 1]
```diff
@@ -188,13 +187,12 @@ do.body.i:                                        ; preds = %do.body.preheader.i
   br i1 %cmp.i23.i, label %if.then.i.i, label %if.else.i24.i
 
 if.then.i.i:                                      ; preds = %do.body.i
-  %conv.i29.i = sext i32 %reader.sroa.13.0.i to i64
```
Comment (goldsteinn) [inline bench/arrow/optimized/hashing.cc.ll:191]: would expect `nneg` here no?

File: bench/cmake/optimized/nghttp2_session.c.ll [hunk 2]
```diff
@@ -14165,26 +14165,28 @@ define internal fastcc range(i32 -902, 1) i32 @session_pack_extension(ptr nounde
...
   %15 = icmp eq i64 %14, -535
-  br i1 %15, label %24, label %16
+  br i1 %15, label %25, label %16
 
 16:                                               ; preds = %2
-  %17 = icmp ugt i64 %14, %spec.select
-  br i1 %17, label %24, label %18
+  %17 = icmp slt i64 %14, 0
+  %18 = icmp samesign ugt i64 %14, %spec.select
+  %or.cond = select i1 %17, i1 true, i1 %18
```
Comment (dtcxzyw) [inline bench/cmake/optimized/nghttp2_session.c.ll:14173]: Missing fold

Comment (goldsteinn) [reply bench/cmake/optimized/nghttp2_session.c.ll:14173]: Maybe the poison generating flag?
https://godbolt.org/z/Wj49jnW5T

**Pull request closed without merging**